tests/macros/t14511.nim

```diff
@@ -0,0 +1,53 @@
+discard """
+  output: "true\n(y: XInt, a: 5)\n(y: XString, b: \"abc\")"
+"""
+
+import macros
+
+block TEST_1:
+  # https://github.com/nim-lang/Nim/issues/14511
+
+  template myPragma() {.pragma.}
+
+  type
+    XType = enum
+      XInt,
+      XString
+    X = object
+      case y {.myPragma.}: XType
+        of XInt:
```

Comment:
Make this `of XInt, XUnused` to see that your version still didn't work properly.

Reply:
What exactly do you mean?
When replacing XInt with XUnused or adding it to (like above), then all tests are still successful. I also cannot see the issue here. It only crashes when in object initialization the wrong XType is passed.

Reply:
```suggestion
        of XInt, XUnused:
```